### PR TITLE
refactor ShardResolversContainerFactoryCreator

### PR DESCRIPTION
**Pull request by axenteoctavian**
## Reasoning behind the pull request
- refactor `ShardResolversContainerFactoryCreator`
- 
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by raduchis — APPROVED** [review]

File: integrationTests/realcomponents/processorRunner.go [hunk 1]
```diff
@@ -49,6 +42,12 @@ import (
 	"github.com/multiversx/mx-chain-go/storage/storageunit"
 	"github.com/multiversx/mx-chain-go/testscommon/headerSigVerifier"
 	"github.com/multiversx/mx-chain-go/update/trigger"
+
+	"github.com/multiversx/mx-chain-core-go/core"
```
Comment (sasurobert) [inline integrationTests/realcomponents/processorRunner.go:46]: goimports

File: testscommon/dataRetriever/shardResolversContainerFactoryMock.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package dataRetriever
+
+import (
+	"github.com/multiversx/mx-chain-go/dataRetriever"
+)
+
+type ShardResolversContainerFactoryMock struct {
```
Comment (sasurobert) [inline testscommon/dataRetriever/shardResolversContainerFactoryMock.go:7]: comments missing.

File: integrationTests/realcomponents/processorRunner.go (hunk 1 above)
Comment (axenteoctavian) [reply integrationTests/realcomponents/processorRunner.go:46]: fixed

File: testscommon/dataRetriever/shardResolversContainerFactoryMock.go (hunk 2 above)
Comment (axenteoctavian) [reply testscommon/dataRetriever/shardResolversContainerFactoryMock.go:8]: fixed

**Review by sasurobert — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Pull request merged**